### PR TITLE
Jordan/dissolve txpool jobs queue

### PR DESCRIPTION
**Pull request by mjfh**
This PR replaces the tx-pool job queue logic with in-place actions. The additional jobs execution layer is not needed, anymore which has been  learned from working with the integration/hive tests.

### CODE REVIEW

File: nimbus/rpc/p2p.nim [hunk 1]
```diff
@@ -248,11 +248,8 @@ proc setupEthRpc*(node: EthereumNode, ctx: EthContext, chain: BaseChainDB, txPoo
       eip155   = chain.currentBlock >= chain.config.eip155Block
       signedTx = signTransaction(tx, acc.privateKey, chain.config.chainId, eip155)
       rlpTx    = rlp.encode(signedTx)
-      res      = txPool.addLocal(signedTx, force = true)
-
-    if res.isErr:
-      raise newException(ValueError, $res.error)
 
+    txPool.add(signedTx)
```
Comment (jangko) [inline nimbus/rpc/p2p.nim:252]: if we have `txpool.add`, why the other PR #1046 ?

**Comment by jangko** [comment]
This is actually a question, because I have not look deeper how txpool works.
I have a case where the sealing-engine/engine-API detect reorg condition. for example the sealing engine is producing side chain block, then realize need to switch chain and go back to forking point.
how the sealing-engine/engine API tell txpool to go back to the forking point?
which call/sequence of calls should be made to txpool?

I'm asking this because there are `setHead`, `smartHead`, and `triggerReorg`.

**Comment by mjfh** [comment]
Good question. Some answer might be found in the `tx_tasks/tx_head.headDiff()` function. I tried to document the different scenarios. From there I might be able to give yoy waht is needed.

The `setHead()` function is internally only and should stay so. The `smartHead()` function is based on exactly the `headDiff()` function, above.

The `triggerReorg()` functionality can probably be represented differently, something like `tx.add @[]` comes to mind. Also, the notion of _dirty buckets_ might have lost its usefulness after removing the job queue.

File: nimbus/utils/tx_pool.nim [hunk 2]
```diff
@@ -573,88 +539,68 @@ proc jobAddTxs*(xp: TxPoolRef; txs: openArray[Transaction]; info = "")
...
-  ## after the cached block chain head is set to the position implied by the
-  ## argument `newHead`. If successful, the txs to add or delete are queued
-  ## on the job queue (run `jobCommit()` to execute) and `true` is returned.
-  ## Otherwise nothing is done and `false` is returned.
-  let rcDiff = xp.headDiff(newHead)
+  ## This function moves the internal head cache (i.e. tx insertion point,
+  ## vmState) and ponts it to a now block on the chain.
+  ##
+  ## In standard mode when argument `blindMode` is `false`, it calculates the
+  ## txs that beed to be added or deleted after moving the insertion point
```
[the remark this answers is not in the document]
Comment (mjfh) [reply nimbus/utils/tx_pool.nim:558]: :)

**Pull request merged**